Disk image (".dmg") creation on macOS fails due to lack of space on device

As seen in [this failed GitHub Actions job](https://github.com/gridsync/gridsync/actions/runs/11961100669/job/33346577548), Gridsync bundles that contain separate Python environments (i.e., one for Gridsync, one for Tahoe-LAFS, and one for Magic-Folder), are now too large to fit into the final macOS disk image, causing `dmgbuild` to fail as follows:

```
python3 -m virtualenv --clear build/venv-dmg
created virtual environment CPython3.12.7.final.0-64 in 738ms
  creator CPython3Posix(dest=/Users/runner/work/gridsync/gridsync/build/venv-dmg, clear=True, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, via=copy, app_data_dir=/Users/runner/Library/Application Support/virtualenv)
    added seed packages: pip==24.3.1
  activators BashActivator,CShellActivator,FishActivator,NushellActivator,PowerShellActivator,PythonActivator
source build/venv-dmg/bin/activate && \
	python3 -m pip install -r requirements/dmgbuild.txt && \
	python3 scripts/call_dmgbuild.py
Collecting dmgbuild==1.6.2 (from -r requirements/dmgbuild.txt (line 7))
  Downloading dmgbuild-1.6.2-py3-none-any.whl (34 kB)
Collecting ds-store==1.3.1 (from -r requirements/dmgbuild.txt (line 11))
  Downloading ds_store-1.3.1-py3-none-any.whl (16 kB)
Collecting mac-alias==2.2.2 (from -r requirements/dmgbuild.txt (line 15))
  Downloading mac_alias-2.2.2-py3-none-any.whl (21 kB)
Installing collected packages: mac-alias, ds-store, dmgbuild
Successfully installed dmgbuild-1.6.2 ds-store-1.3.1 mac-alias-2.2.2
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/QtWidgets.abi3.so: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/QtGui.abi3.so: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/QtOpenGL.abi3.so: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/plugins/platforms/libqminimal.dylib: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/plugins/platforms/libqcocoa.dylib: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/plugins/styles/libqmacstyle.dylib: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/plugins/iconengines/libqsvgicon.dylib: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/plugins/imageformats/libqgif.dylib: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/plugins/imageformats/libqwbmp.dylib: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/plugins/imageformats/libqwebp.dylib: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/plugins/imageformats/libqico.dylib: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/plugins/imageformats/libqmacheif.dylib: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/plugins/imageformats/libqjpeg.dylib: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/plugins/imageformats/libqtiff.dylib: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/plugins/imageformats/libqsvg.dylib: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/plugins/imageformats/libqpdf.dylib: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/plugins/imageformats/libqicns.dylib: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/plugins/imageformats/libqtga.dylib: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/plugins/imageformats/libqmacjp2.dylib: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/plugins/generic/libqtuiotouchplugin.dylib: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/lib/QtPdf.framework/Versions/A/QtPdf: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/lib/QtGui.framework/Versions/A/QtGui: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/lib/QtDBus.framework/Versions/A/QtDBus: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/lib/QtCharts.framework: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/lib/QtCore.framework: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/lib/QtOpenGL.framework: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/lib/QtOpenGLWidgets.framework: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/lib/QtWidgets.framework: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/lib/QtNetwork.framework: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/lib/QtSvg.framework: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/Qt6/translations: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/QtCore.abi3.so: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/QtCharts.abi3.so: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/QtDBus.abi3.so: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/sip.cpython-310-darwin.so: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Frameworks/PyQt6/QtOpenGLWidgets.abi3.so: No space left on device
ditto: /Volumes/Gridsync/Gridsync.app/Contents/Info.plist: No space left on device
hdiutil: couldn't eject "disk2" - Resource busy
Traceback (most recent call last):
  File "/Users/runner/work/gridsync/gridsync/build/venv-dmg/bin/dmgbuild", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/Users/runner/work/gridsync/gridsync/build/venv-dmg/lib/python3.12/site-packages/dmgbuild/__main__.py", line 56, in main
    build_dmg(
  File "/Users/runner/work/gridsync/gridsync/build/venv-dmg/lib/python3.12/site-packages/dmgbuild/core.py", line 709, in build_dmg
    os.symlink(target, name_in_image)
OSError: [Errno 28] No space left on device: '/Applications' -> '/Volumes/Gridsync/Applications'
Traceback (most recent call last):
Hashing (SHA256)...
  File "/Users/runner/work/gridsync/gridsync/scripts/call_dmgbuild.py", line 28, in <module>
    with open(path, 'rb') as f:
         ^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'dist/Gridsync.dmg'
make: *** [dmg] Error 1
```

The size of the image generated by `dmgbuild` should be increased (or sized dynamically?) in order to account for increases in the overall size of the application bundle.

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -108,7 +108,7 @@ jobs:
     - name: sha256sum
       run: python3 scripts/sha256sum.py dist/*.*
     - name: Upload artifacts
-      uses: actions/upload-artifact@v2
+      uses: actions/upload-artifact@v4
       with:
         path: dist/Gridsync.AppImage
         name: Gridsync-${{ matrix.os }}-${{ matrix.qt }}.AppImage
@@ -176,7 +176,7 @@ jobs:
     - name: sha256sum
       run: python3 scripts/sha256sum.py dist/*.*
     - name: Upload artifacts
-      uses: actions/upload-artifact@v2
+      uses: actions/upload-artifact@v4
       with:
         path: dist/Gridsync.dmg
         name: Gridsync-${{ matrix.os }}-${{ matrix.qt }}.dmg
@@ -224,7 +224,7 @@ jobs:
       run: |
         .\make.bat test-determinism
     - name: Upload zipfiles
-      uses: actions/upload-artifact@v2
+      uses: actions/upload-artifact@v4
       if: failure()
       with:
         name: zipfiles.zip
@@ -235,7 +235,7 @@ jobs:
     - name: sha256sum
       run: python3 scripts/sha256sum.py .\dist\*.*
     - name: Upload artifacts
-      uses: actions/upload-artifact@v2
+      uses: actions/upload-artifact@v4
       with:
         path: .\dist\Gridsync-setup.exe
         name: Gridsync-${{ matrix.os }}-${{ matrix.qt }}-setup.exe

--- a/misc/dmgbuild_settings.py
+++ b/misc/dmgbuild_settings.py
@@ -60,7 +60,7 @@ def icon_from_app(app_path):
 format = defines.get('format', 'UDBZ')
 
 # Volume size (must be large enough for your files)
-size = defines.get('size', '150M')
+size = defines.get('size', '250M')
 
 # Files to include
 files = [ application ]

